components/definition/cellRules/sections/mechanisms/NetPyNEMechanism.js

```diff
@@ -1,107 +1,65 @@
 import React, { Component } from 'react';
 import SelectField from 'material-ui/SelectField';
-import MenuItem from 'material-ui/MenuItem';
 import TextField from 'material-ui/TextField';
-import Tooltip from 'material-ui/internal/Tooltip';
-import FlatButton from 'material-ui/FlatButton';
-import RaisedButton from 'material-ui/RaisedButton';
-import Toggle from 'material-ui/Toggle';
-import FontIcon from 'material-ui/FontIcon';
-import CardText from 'material-ui/Card';
 import Utils from '../../../../../Utils';
-import { BottomNavigation, BottomNavigationItem } from 'material-ui/BottomNavigation';
+import NetPyNEField from '../../../../general/NetPyNEField';
 
 var PythonControlledCapability = require('../../../../../../../js/communication/geppettoJupyter/PythonControlledCapability');
 var PythonControlledTextField = PythonControlledCapability.createPythonControlledControl(TextField);
-var PythonControlledSelectField = PythonControlledCapability.createPythonControlledControl(SelectField);
+var PythonMethodControlledSelectField = PythonControlledCapability.createPythonControlledControlWithPythonDataFetch(SelectField);
 
 export default class NetPyNEMechanism extends React.Component {
 
   constructor(props) {
     super(props);
-    var _this = this;
     this.state = {
-      currentName: props.name,
-      selectedIndex: 0,
-      sectionId: "General"
+      currentMech: props.name,
```

Comment:
We usually use currentName, right? Let's keep the same convention...

Reply:
ok. will update